MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,226 @@
+#    This file is part of MDANSE.
+#
+#    MDANSE is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import collections
+import copy
+import json
+
+import numpy as np
+
+import h5py
+from MDANSE.Framework.Formats import HDFFormat
+
+from MDANSE.Chemistry.ChemicalEntity import AtomGroup
+from MDANSE.Framework.Jobs.IJob import IJob
+from MDANSE.Mathematics.Signal import filter_map
+from MDANSE.MolecularDynamics.Configuration import RealConfiguration
+from MDANSE.MolecularDynamics.Trajectory import sorted_atoms, TrajectoryWriter
+from MDANSE.MLogging import LOG
+
+
+class TrajectoryFilter(IJob):
+    """
+    Design and apply a filter for the atomic trajectories.
+    """
+
+    label = "Trajectory Filter"
+
+    category = (
+        "Analysis",
+        "Dynamics",
+    )
+
+    ancestor = ["hdf_trajectory", "molecular_viewer"]
+
+    settings = collections.OrderedDict()
+    settings["trajectory"] = ("HDFTrajectoryConfigurator", {})
+    settings["frames"] = (
+        "CorrelationFramesConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
```

Comment:
Since this isn't a job that calculates a correlation function, you can change this to the FramesConfigurator.